### PR TITLE
Fix building issue due to "$<LINK_LANGUAGE:CXX>" and "SHELL:"

### PR DESCRIPTION
**Pull request by tomsang**
Replace  $<LINK_LANGUAGE:CXX> with 1; 
Remove "SHELL:"

### CODE REVIEW

**Comment by pfultz2** [comment]
We need the linking flags to build a code object so they can't be removed. I think it would be better to do:

```cmake
string(REPLACE "$<LINK_LANGUAGE:CXX>" "1" HIP_COMPILER_FLAGS "${HIP_COMPILER_FLAGS}")
```

This will replace `$<LINK_LANGUAGE:CXX>` with "1" so it will always be treated as true inside the genex.

File: cmake/TargetFlags.cmake
Comment (pfultz2) [inline cmake/TargetFlags.cmake:56]: This wont get the complete flags as it wont transitively traverse the targets, so it will miss the flags from `hip::host`.

Comment (tomsang) [reply cmake/TargetFlags.cmake]: Fixed

**Review by pfultz2 — APPROVED** [review]

**Comment by tomsang** [comment]
!verify full

**Comment by atamazov** [comment]
@tomsang I am pushing this though our CI. Will merge after all tests passed. Ping me if I will not update or merge this PR in a couple of days.

**Comment by atamazov** [comment]
@tomsang Please merge recent develop to your branch. This is to resolve CI errors. I can't do that for you. Then I will continue with https://github.com/ROCmSoftwarePlatform/MIOpen/pull/983#issuecomment-864298056

**Comment by atamazov** [comment]
@tomsang Ping

**Comment by tomsang** [comment]
@atamazov 
I find I am not authorized to merge it.
Could you help find an authorized person to merge it ?

Thanks!

**Comment by atamazov** [comment]
@tomsang Yes, you can't merge this PR into develop. I can do that. But first I have to make sure that this PR passes all the necessary CI checks. However, it fails due to problems that old develop (that you've based your branch on) has.

In order to resolve these problems:

> Please merge recent develop to your branch. 

How to do this:
```bash
git fetch https://github.com/ROCmSoftwarePlatform/MIOpen develop
git checkout develop
git pull https://github.com/ROCmSoftwarePlatform/MIOpen develop
git checkout fix_miopen_compile_flags
git merge develop
#
# if merge conflicts, resolve them and then `git commit`
#
git push YOUR_REPOSITIRY fix_miopen_compile_flags
```
The only permissions you need is pulling develop from this repo, which should be granted for everyone. After your branch updated in your repo, our CI will start the testing job (or I will do that manually, if necessary).

Please let me know if something is not working or unclear.

Please also let us know if you would like to obtain the collaborator permissions in this repo.

**Comment by tomsang** [comment]
@atamazov 
I have done that.

**Comment by atamazov** [comment]
CI job started.

**Comment by atamazov** [comment]
CI job passed all tests.

**Comment by atamazov** [comment]
@tomsang Merged. Thanks for your contribution!

Could you please update the PR description with some reasoning why this cheng is needed?

**Pull request merged**
